### ISSUE TITLE
Compile and run end-to-end tests in Java

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [general-language-syntax/GLS#354]
Follows #55. A Java runtime should execute each test's `expected.java` and verify the output is the same as `output.txt`.

**Comment by JoshuaKGoldberg** [comment]
This one was oddly tricky. See later commits in https://github.com/general-language-syntax/GLS/pull/505. I can get it running locally on Windows but on Travis, even with Windows builds, it errors out.

Java versions (per [this build](https://travis-ci.org/general-language-syntax/GLS/builds/447291185?utm_source=github_status&utm_medium=notification)):

```
java version "1.8.0_151"
Java(TM) SE Runtime Environment (build 1.8.0_151-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.151-b12, mixed mode)
```

Error as per [this build](https://travis-ci.org/general-language-syntax/GLS/builds/447285362?utm_source=github_status&utm_medium=notification):

```
Error: Error: Could not find or load main class .home.travis.build.general-language-syntax.GLS.test.end-to-end.Arrays.index.java
```

**Comment by JoshuaKGoldberg** [comment]
Fixed by #599! ✨
